libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewHolder.kt

```diff
@@ -0,0 +1,42 @@
+package org.wordpress.android.imageeditor.preview
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.ProgressBar
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.recyclerview.widget.RecyclerView
+import org.wordpress.android.imageeditor.R
+import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageData
+import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageUiState
+import org.wordpress.android.imageeditor.utils.UiHelpers
+
+class PreviewImageViewHolder(
+    val view: View,
+    private val loadIntoImageViewWithResultListener: (ImageData, ImageView, Int) -> Unit
+) : RecyclerView.ViewHolder(view) {
+    private val previewImageView = itemView.findViewById<ImageView>(R.id.previewImageView)
+    private val progressBar = itemView.findViewById<ProgressBar>(R.id.progressBar)
+    private val errorLayout = itemView.findViewById<ConstraintLayout>(R.id.errorLayout)
+
+    fun onBind(uiState: ImageUiState) {
+        loadIntoImageViewWithResultListener.invoke(uiState.data, previewImageView, adapterPosition)
+        UiHelpers.updateVisibility(progressBar, uiState.progressBarVisible)
+        UiHelpers.updateVisibility(errorLayout, uiState.retryLayoutVisible)
+    }
+
+    companion object {
+        fun create(
```

Comment:
Have you considered using [this approach](https://github.com/wordpress-mobile/WordPress-Android/blob/2b681ae6d87f837c2eab29a99a428ce2a7a875fc/WordPress/src/main/java/org/wordpress/android/ui/main/ActionListItemViewHolder.kt#L18) instead of the companion object? Is there any advantage?

Reply:
I was used to companion object approach from [Android Architecture Blueprints sample](https://github.com/android/architecture-samples/blob/master/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksAdapter.kt#L53-L60) and it just creeped in. There was no added advantage so used suggested approach here: bf8f911.